lib/streamlit/runtime/caching/cache_utils.py

```diff
@@ -93,6 +165,32 @@ class CachedResult:
     sidebar_id: str
 
 
+@dataclass
+class InitialCachedResults:
+    """Widgets called by a cache-decorated function, and a mapping of the
+    widget-derived cache key to the final results of executing the function.
+    """
+
+    widget_ids: Set[str]
+    results: Dict[str, CachedResult]
```

Comment:
`results` is not an intuitive name to me - I know it's more verbose, but what about something like `results_by_widgets_hash` instead?

I also wonder if this class could have a more descriptive name as well  - it's not really "initial cached results", it's more "set of all thus-far computed results for a given function". What about `CachedResultSet` or something? `CachedResultContainer`?

Reply:
@tconkling 
What do you think about my previous idea of `PartialCachedResults`? Or how about something like `ArgKeyCachedResult`, to distinguish it from the full results that incorporate the widget state derived key?